flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/adaptive/AdaptiveScheduler.java

```diff
@@ -556,12 +559,21 @@ public AdaptiveScheduler(
         deploymentTimeMetrics =
                 new DeploymentStateTimeMetrics(jobGraph.getJobType(), jobStatusMetricsSettings);
 
+        this.executionStateMetricsRegistrars = new ArrayList<>(2);
+        this.executionStateMetricsRegistrars.add(
+                new DeploymentStateTimeMetrics(jobGraph.getJobType(), jobStatusMetricsSettings));
+        if (jobGraph.getJobType() == JobType.STREAMING) {
```

Comment:
why only for streaming?

Reply:
The thinking here is that this event doesn't make sense for batch, where it's natural that not all tasks are started/running from the beginning. 

In streaming on the other hand, this tells you when the job has fully recovered.

I will explain that in the java doc and/or in the documentation.

Reply:
Actually this check can be dropped, as `AdaptiveScheduler` can be only used for streaming jobs anyway (this is already asserted in the `AdaptiveScheduler#assertPreconditions`). 

I've mentioned that this event is only reported for streaming jobs in the docs.